code/modules/admin/admin_verbs.dm

```diff
@@ -79,59 +79,85 @@
 	message_admins("[ADMIN_TPMONTY(M)] has turned stealth mode [M.client.holder.fakekey ? "on - [M.client.holder.fakekey]" : "off"].")
 
 
-/datum/admins/proc/change_key(mob/M in GLOB.alive_mob_list)
+/datum/admins/proc/give_mob(mob/living/L in GLOB.mob_living_list)
 	set category = null
-	set name = "Change CKey"
+	set name = "Give Mob"
 
 	if(!check_rights(R_ADMIN))
 		return
 
-	var/new_ckey = input("Enter new ckey:", "Change CKey") as null|text
+	var/mob/M
+	switch(input("Who do you want to give it to:", "Give Mob") as null|anything in list("Key", "Mob"))
+		if("Key")
+			var/client/C = input("Please, select a key.", "Give Mob") as null|anything in sortKey(GLOB.clients)
+			if(!C)
+				return
+			M = C.mob
+		if("Mob")
+			var/mob/N = input("Please, select a mob.", "Give Mob") as null|anything in sortNames(GLOB.mob_list)
+			if(!N)
+				return
+			M = N
 
-	if(!new_ckey)
+	if(isliving(M) && M.client && alert("[key_name(M)] is already playing, do you want to proceed?", "Give Mob", "Yes", "No") != "Yes")
 		return
 
-	M.ghostize(FALSE)
-	M.ckey = ckey(new_ckey)
-	if(M.client)
-		M.client.change_view(world.view)
+	if(!istype(L))
+		return
+
+	log_admin("[key_name(usr)] gave [key_name(L)] to [key_name(M)].")
+	message_admins("[ADMIN_TPMONTY(usr)] gave [ADMIN_TPMONTY(L)] to [ADMIN_TPMONTY(M)].")
 
-	log_admin("[key_name(usr)] changed [M.name] ckey to [new_ckey].")
-	message_admins("[ADMIN_TPMONTY(usr)] changed [M.name] ckey to [new_ckey].")
+	L.take_over(M, TRUE)
 
 
-/datum/admins/proc/change_key_panel()
+/datum/admins/proc/give_mob_panel()
 	set category = "Admin"
-	set name = "Change CKey Mob"
+	set name = "Give  Mob"
 
 	if(!check_rights(R_ADMIN))
 		return
 
-	var/mob/M
-	switch(input("Change by:", "Change CKey") as null|anything in list("Key", "Mob"))
+	var/mob/living/L
+	switch(input("Who do you want to give:", "Give Mob") as null|anything in list("Key", "Mob"))
 		if("Key")
-			var/client/C = input("Please, select a key.", "Change CKey") as null|anything in sortKey(GLOB.clients)
-			if(!C)
+			var/client/C = input("Please, select a key.", "Give Mob") as null|anything in sortKey(GLOB.clients)
+			if(!C || !isliving(C.mob))
```

Comment:
```suggestion
			if(!isliving(C?.mob))
```

Reply:
Though is this a check we want, rather one we'd like for the runtime to appear?

Reply:
No